doc/ir-semantics.text

```diff
@@ -111,18 +111,23 @@ Update (partial) function, returns an H:
 
 ## Osr transformation of environment
 
-  materialize P H E (mut x = e) = (x ↦ (a ↦ v))
-    where a - fresh
-      and v := eval P H E e
-  materialize P H E _           = ()
-
-  osr-env P H E M (const x = e) = (x ↦ val v)
+  build-env H E (const x = e) = (∅, [x ↦ val v])
     where v := eval P H E e
-  osr-env P H E M (mut x)       = (x ↦ ⊥)
-  osr-env P H E M (mut x = &y)  = (x ↦ ref a)
+  build-env H E (mut x)       = ([a ↦ ⊥], [x ↦ ref a])
+    where a fresh
+  build-env H E (mut x = e)   = ([a ↦ v], [x ↦ ref a])
+    where a fresh
+      and v := eval P H E e
+  build-env H E (mut x = &y)  = (∅, [x ↦ ref a])
     where E ∋ (y ↦ ref a)
-  osr-env P H E M (mut x = e)   = (x ↦ ref a)
-    where M ∋ (x ↦ (a ↦ _))
+
+  extend-env H E def := (H, E) + build-env H E def
```

Comment:
`def` was never really defined. Right now it's the same as `osr-map`. Should we just rename `osr-map` to `def` or something else?